<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="utf-8"/>
	<title>Taming the BEAST Down Under</title>
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	
	<!-- RSS feed -->
	<link rel="alternate" type="application/rss+xml" title="Taming the BEAST" href="https://taming-the-beast.org//feed.news.xml">
	<link rel="alternate" type="application/rss+xml" title="Taming the BEAST" href="https://taming-the-beast.org//feed.tutorials.xml">

	
    <!-- Favicon -->
    <link rel="shortcut icon" href="/images/favicon.ico">	

    <!-- Our style -->
    <link rel="stylesheet" type="text/css" href="/css/style.css">

    <!-- JQuery and Bootstrap JS (necessary for collapsing navbar) -->
    <script src="/js/jquery-3.6.0.min.js"></script>
    <script src="/bootstrap/js/bootstrap.bundle.min.js"></script>

    <!-- Bootstrap Icons -->
    <link rel="stylesheet" type="text/css" href="/bootstrap/icons/bootstrap-icons.css">

    <!-- KaTeX -->
    <link rel="stylesheet" href="/css/katex.css">
    <script src="/js/katex.min.js"></script>

</head>


<body>
	
	<div id="header">

  <nav class="navbar navbar-expand-lg navbar-light bg-light">
    
    <a class="navbar-brand" href="/">
      <img src="/images/logo_bw.svg" style="height:2em">Taming the BEAST
    </a>
    
  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon"></span>
  </button>
  <div class="collapse navbar-collapse" id="navbarNav">
    <ul class="navbar-nav">
      <li class="nav-item active">
        <a class="nav-link " aria-current="page" href="/news/">news</a>
      </li>
      <li class="nav-item">
        <a class="nav-link  active " href="/workshops/">workshops</a>
      </li>
      <li class="nav-item">
        <a class="nav-link " href="/tutorials/">tutorials</a>
      </li>
      <li class="nav-item">
        <a class="nav-link " href="/contribute/">contribute</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" type="application/rss+xml" href="/feed.news.xml">
          <i class="bi bi-rss"></i>
        </a>
      </li>
    </ul>
  </div>
</nav>

</div>

	
	<div class="container">	
		
	<div class="row">
	<div class="col-md-10">
		<div class="bigtitle titlebox">
			<a class="off" target="_blank" href="https://sydney.edu.au/science/biology/meep/workshops/">Taming the BEAST Down Under</a>
		</div>
		<p>
		<div class="head">
			

            	        <a class="off" target="_blank" href="http://www.google.com/maps/place/-33.8873,151.1835">
                          
                            Charles Perkins Centre, University of Sydney,
                          
                          Sydney, Australia</a>
                        
                        
		</div>
		<div class="smallhead">
                  
		  February 17, 2019-February 22, 2019
                  
</li> 
		</div>
	</div>		

	<div class="col-md-2 text-right">
    	<img width=160 class="media-object" src="/images/workshops/Taming-the-BEAST-Down-Under-Logo.png">
	</div>		
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-12">
		<h1 id="workshop-programme">Workshop Programme</h1>

<p>Where possible, links have been added from the programme below to the corresponding tutorials and lecture slides.
Additional links will be added as they become available.</p>

<table>
<thead>

<tr>
<th></th>
<th> Sunday <br /> 2019/02/17 </th>
<th> Monday <br /> 2019/02/18</th>
<th> Tuesday <br /> 2019/02/19 </th>
<th> Wednesday <br /> 2019/02/20 </th>
<th> Thursday <br /> 2019/02/21 </th>
<th> Friday <br /> 2019/02/22 </th>
</tr>

</thead>

<tbody>

<tr>
<td> 09:00</td>
<td></td>
<td> <b>Alexei Drummond</b> <br /> <a href="https://github.com/Taming-the-BEAST/Taming-the-BEAST-2019-Sydney-Lectures/raw/master/Day1_Alexei_BayesianPhylogenetics.pdf">Bayesian Phylogenetics and BEAST</a> </td>
<td> <b>Remco Bouckaert</b> <br /> <a href="https://github.com/Taming-the-BEAST/Taming-the-BEAST-2019-Sydney-Lectures/raw/master/Day2_ModelSelection.pdf">Phylogenetic model selection and averaging</a> </td>
<td> <b>Tim Vaughan, Jana Huisman, Nicola M&uuml;ller</b> <br /> Bayesian Phylodynamics: <a href="https://tgvaughan.github.io/TTB_Lectures/Phylodynamics">Skylines</a>, <a href="https://github.com/Taming-the-BEAST/Taming-the-BEAST-2019-Sydney-Lectures/raw/master/Day3_Jana_FBD.pdf">Fossilization</a> and <a href="https://slides.com/nicolamueller/20190218ttbstrutured#/20190218ttbstrutured">Structure</a> </td>
<td> <b>Mike Lee</b> <br /> Bayesian phylogenetics for phenotypic and palaeontological datasets (<a href="https://github.com/Taming-the-BEAST/Taming-the-BEAST-2019-Sydney-Lectures/raw/master/Day4_MikeLee_SydneyBayes1.pdf">part 1</a>,
<a href="https://github.com/Taming-the-BEAST/Taming-the-BEAST-2019-Sydney-Lectures/raw/master/Day4_MikeLee_SydneyBayes2.pdf">part 2</a>)</td>
<td> <b>Sebastián Duchêne</b> <br /> <a href="https://github.com/Taming-the-BEAST/Taming-the-BEAST-2019-Sydney-Lectures/raw/master/Day5_SebastianDuchene_TipDating.pdf">Tip-dating and phylodynamics</a> </td>
</tr>

<tr>
<td> 11:00 </td>
<td></td>
<td> Morning Tea </td>
<td> Morning Tea </td>
<td> Morning Tea </td>
<td> Morning Tea </td>
<td> Morning Tea </td>
</tr>


<tr>
<td> 11:30 </td>
<td></td>
<td rowspan="2"> <b>Tim Vaughan</b>
    <br /> First Steps in BEAST&nbsp;2<br />
    (<a href="https://tgvaughan.github.io/TTB_Lectures/FirstSteps/">lecture</a>, <a href="https://taming-the-beast.org/tutorials/Introduction-to-BEAST2/">tutorial</a>) </td>
<td rowspan="2"> <b>Remco Bouckaert</b> <br /> Substitution model averaging <br />
    (<a href="https://taming-the-beast.org/tutorials/Substitution-model-averaging/">tutorial</a>) </td>
<td> <b>Mike Lee</b> <br /> Arid biomes as a source, not sink, for Australian lizard biodiversity
<br /> (public lecture) </td>
<td rowspan="2"> <b>Jana Huisman</b> <br /> Fossilized birth-death model analyses <br /> (<a href="https://taming-the-beast.org/tutorials/FBD-tutorial/">tutorial</a>) </td>
<td rowspan="2"> <b>Nicola Müller</b> <br /> Isolation with Migration using Coupled MCMC <br /> (<a href="https://taming-the-beast.org/tutorials/AIM-Tutorial">tutorial</a>) </td>
</tr>

<tr>
<td> 12:30 </td>
<td></td>
<td rowspan="10" style="border: solid 1px gray"> Free afternoon<br />(packed lunch provided)</td>
</tr>

<tr>
<td> 13:00 </td>
<td></td>
<td>Lunch</td>
<td>Lunch</td>
<td>Lunch</td>
<td>Lunch</td>
</tr>

<tr>
<td> 14:00 </td>
<td></td>
<td rowspan="2"> <b>Jana Huisman</b> <br /> Prior Selection &amp; Troubleshooting <br /> (<a href="https://github.com/Taming-the-BEAST/Taming-the-BEAST-2019-Sydney-Lectures/raw/master/Priors_and_trouble.pdf">lecture</a>, <a href="https://taming-the-beast.org/tutorials/Prior-selection/">tutorial</a>)</td>
<td rowspan="2"> <b>Simon Ho </b> <br /> Molecular Dating (<a href="https://github.com/Taming-the-BEAST/Taming-the-BEAST-2019-Sydney-Lectures/raw/master/Day2_MolecularDating.pdf">lecture</a>)</td>
<td> <b>David Duchêne</b> <br /> Species trees and gene trees
(<a href="https://github.com/Taming-the-BEAST/Taming-the-BEAST-2019-Sydney-Lectures/raw/master/Day4_SpeciesGeneTreesDDuchene.pdf">lecture</a>)</td>
<td rowspan="2"> <b>Tim Vaughan</b> <br /> Bacterial genomics<br />
(<a href="https://tgvaughan.github.io/TTB_Lectures/ARGs/">lecture</a>, <a href="https://taming-the-beast.org/tutorials/Bacter-Tutorial/">tutorial</a>)</td>
</tr>

<tr>
<td> 15:00 </td>
<td></td>
<td rowspan="3"> <b>Jérémie Sciré, Nicola Müller</b> <br />
    Analyzing structured populations under birth-death and coalescent models <br />
    (<a href="https://taming-the-beast.org/tutorials/Structured-birth-death-model/">tutorial 1</a>, <a href="https://taming-the-beast.org/tutorials/Mascot-Tutorial/">tutorial 2</a>)
</td>
</tr>


<tr>
<td> 15:30 </td>
<td></td>
<td> Afternoon Tea </td>
<td> Afternoon Tea </td>
<td> Afternoon Tea </td>
</tr>

<tr>
<td> 16:00 </td>
<td rowspan="5" style="border: solid 1px gray">
Welcome and registration<br /> <i>(Courtyard Cafe,<br />University of Sydney)</i></td>
<td rowspan="3"> <b>Nicola Müller</b> <br /> Skyline Plot Tutorial <br />
(<a href="https://slides.com/nicolamueller/skyline-plots/">lecture</a>,
<a href="https://taming-the-beast.org/tutorials/Skyline-plots/">tutorial</a>)</td>
<td rowspan="3"> <b>Cara Van Der Wal,  Daej Arab, Yi-Kai Tea</b> <br /> Divergence time estimation <br /> (<a href="https://taming-the-beast.org/tutorials/CladeAge-Tutorial/">tutorial</a>)</td>
<td rowspan="5" style="border: solid 1px gray">
Departure</td>
</tr>

<tr>
<td> 16:30 </td>
<td> </td>
</tr>

<tr>
<td> 17:00 </td>
</tr>

<tr>
<td> 18:00 </td>
<td> </td>
<td> </td>
<td rowspan="2">
Buffet dinner cruise<br />
<i>(Sydney Harbour, Depart from King St Wharf)</i></td>
</tr>

<tr>
<td> 18:30 </td>
<td> Dinner<br /> <i>(Thai Pothong, Newtown)</i></td>
<td> </td>
</tr>

</tbody>
</table>

<h1 id="required-software">Required Software</h1>

<p><strong>You are required to bring your own laptop to the workshop.</strong> Power strips are available for each participant to keep their devices charged during the workshop. The software below <strong>must</strong> be installed prior to arrival.</p>

<ul>
  <li><strong>Java 8 or higher:</strong> This is required by BEAST 2 <a href="http://java.com/download">(download)</a>.</li>
  <li><strong>BEAST 2:</strong> BEAST 2 is a cross-platform program for Bayesian phylogenetic analysis of molecular sequences. We will be using this software to perform analyses on the data brought to the workshop. We will be using <strong>BEAST v2.5.2</strong> for the workshop <a href="http://beast2.org/">(download)</a>.</li>
  <li><strong>FigTree:</strong> FigTree is a graphical viewer of phylogenetic trees, specifically designed to display summarised and annotated trees produced by BEAST <a href="http://beast.community/figtree">(download)</a>.</li>
  <li><strong>Tracer:</strong> Tracer is a program for analysing the trace files generated by Bayesian MCMC runs. It can be used to check mixing, ESS, parameter correlations, etc. <a href="http://beast.community/tracer">(download)</a>.</li>
  <li><strong>BEAGLE (optional):</strong> BEAGLE is a high-performance library that can perform the core calculations in BEAST2 (and many other phylogenetics packages) much faster. BEAGLE isn’t necessary for the workshop, but it will speed up your analyses <a href="https://github.com/beagle-dev/beagle-lib">(download)</a>.</li>
  <li><strong>R:</strong> R is a free software environment that provides a wide variety of statistical (linear and nonlinear modelling, classical statistical tests, time-series analysis, classification, clustering) and graphical techniques, and is highly extensible <a href="https://www.r-project.org/">(download)</a>.</li>
  <li><strong>RStudio (optional):</strong> RStudio provides a friendly front-end for R <a href="https://www.rstudio.com/">(download)</a>.</li>
</ul>

<h1 id="pre-workshop-recommended-reading">Pre-workshop recommended reading</h1>

<h3 id="statistics-and-mathematics-primers">Statistics and mathematics primers</h3>

<p>We are aware that some of the workshop participants do not have a mathematical background and may have difficulty understanding some of the theory. The two links below can help to refresh the basic concepts, but are by no means meant to be comprehensive resources.</p>

<ul>
  <li><a href="http://students.brown.edu/seeing-theory/index.html">Seeing Theory</a>: This website gives a very brief, but pretty and easy to understand introduction to many of the fundamental concepts in statistics.</li>
  <li>Ziheng Yang’s <a href="http://abacus.gene.ucl.ac.uk/PPS/PrimerProbabilityStatistics.pdf">statistics primer</a>: Gives a more in-depth introduction to the topic.</li>
</ul>

<h3 id="phylogenetics-and-phylodynamics-books">Phylogenetics and phylodynamics books</h3>

<p>The books below are not all focused on BEAST or even Bayesian phylogenetics. However, each of the books contains at least one background chapter on the theory of molecular evolution, as well as more in-depth chapters on the models implemented in BEAST and BEAST 2. Most of the books also contain at least some discussion of Bayesian inference.</p>

<ul>
  <li><a href="https://www.beast2.org/book/">Bayesian Evolutionary Analysis with BEAST</a>, <em>Alexei J. Drummond, Remco R. Bouckaert</em></li>
  <li><a href="http://abacus.gene.ucl.ac.uk/MESA/">Molecular Evolution: A Statistical Approach</a>, <em>Ziheng Yang</em></li>
  <li><a href="http://www.cambridge.org/catalogue/catalogue.asp?isbn=9780521877107">The Phylogenetic Handbook</a>, <em>Philippe Lemey, Marco Salemi, Anne-Mieke Vandamme</em></li>
  <li><a href="http://eu.wiley.com/WileyCDA/WileyTitle/productCd-0865428891.html">Molecular Evolution: A Phylogenetic Approach</a>, <em>Roderick D.M. Page, Edward C. Holmes</em></li>
  <li><a href="https://www.amazon.co.uk/Inferring-Phylogenies-Joseph-Felsenstein/dp/0878931775">Inferring Phylogenies</a>, <em>Joseph Felsenstein</em></li>
</ul>

<p>In addition, the following review papers might be useful:</p>

<ul>
  <li><a href="https://www.nature.com/articles/nrg.2015.8">Bayesian molecular clock dating of species divergences in the genomics era</a>, <em>Mario dos Reis, Phillip Donoghue, Ziheng Yang</em></li>
  <li><a href="https://www.nature.com/articles/nrg2583">Evolutionary analysis of the dynamics of viral infectious disease</a>, <em>Oliver Pybus, Andrew Rambaut</em></li>
  <li><a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/brv.12390">Bayesian molecular dating: opening up the black box</a>, <em>Lindell Bromham, Sebastián Duchêne, Xia Hua, Andrew M. Ritchie, David A. Duchêne, Simon Y. W. Ho</em></li>
</ul>

	</div>
</div>

	
	</div>

	<div class="footerspacer"></div>

<div class="footer">
  <div class="container">
    <div class="row">
      <div class="col-md-6">
        <ul class="list-group list-group-horizontal">
        <li class="footernav list-group-item"><a class="foot" href="/about/">about</a></li>
        <li class="footernav list-group-item"><a class="foot" href="/contact/">contact</a></li>
        <li class="footernav list-group-item"><a class="foot" href="/license/">license</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

	
	<!--div id="footer"><span style="display:none">foo</span></div-->
		
</body>
</html>

