<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="utf-8"/>
	<title>Squamish BEAST Tamers</title>
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	
	<!-- RSS feed -->
	<link rel="alternate" type="application/rss+xml" title="Taming the BEAST" href="https://taming-the-beast.org//feed.news.xml">
	<link rel="alternate" type="application/rss+xml" title="Taming the BEAST" href="https://taming-the-beast.org//feed.tutorials.xml">

	
    <!-- Favicon -->
    <link rel="shortcut icon" href="/images/favicon.ico">	

    <!-- Our style -->
    <link rel="stylesheet" type="text/css" href="/css/style.css">

    <!-- JQuery and Bootstrap JS (necessary for collapsing navbar) -->
    <script src="/js/jquery-3.6.0.min.js"></script>
    <script src="/bootstrap/js/bootstrap.bundle.min.js"></script>

    <!-- Bootstrap Icons -->
    <link rel="stylesheet" type="text/css" href="/bootstrap/icons/bootstrap-icons.css">

    <!-- KaTeX -->
    <link rel="stylesheet" href="/css/katex.css">
    <script src="/js/katex.min.js"></script>

</head>


<body>
	
	<div id="header">

  <nav class="navbar navbar-expand-lg navbar-light bg-light">
    
    <a class="navbar-brand" href="/">
      <img src="/images/logo_bw.svg" style="height:2em">Taming the BEAST
    </a>
    
  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon"></span>
  </button>
  <div class="collapse navbar-collapse" id="navbarNav">
    <ul class="navbar-nav">
      <li class="nav-item active">
        <a class="nav-link " aria-current="page" href="/news/">news</a>
      </li>
      <li class="nav-item">
        <a class="nav-link  active " href="/workshops/">workshops</a>
      </li>
      <li class="nav-item">
        <a class="nav-link " href="/tutorials/">tutorials</a>
      </li>
      <li class="nav-item">
        <a class="nav-link " href="/contribute/">contribute</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" type="application/rss+xml" href="/feed.news.xml">
          <i class="bi bi-rss"></i>
        </a>
      </li>
    </ul>
  </div>
</nav>

</div>

	
	<div class="container">	
		
	<div class="row">
	<div class="col-md-10">
		<div class="bigtitle titlebox">
			<a class="off" target="_blank" href="https://ttbcanada.ca/">Squamish BEAST Tamers</a>
		</div>
		<p>
		<div class="head">
			

                          
                          Squamish, British Columbia

                        
		</div>
		<div class="smallhead">
                  
		  August 14, 2023-August 18, 2023
                  
</li> 
		</div>
	</div>		

	<div class="col-md-2 text-right">
    	<img width=160 class="media-object" src="/images/workshops/Taming-the-BEAST-SBT-Logo.png">
	</div>		
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-12">
		<h1 id="workshop-programme">Workshop Programme</h1>

<table style="width:40%">
<tbody>
<tr>
<td class="invited-lecture"><b>Invited talks</b></td>
<td class="lecture"><b>Lectures</b></td>
<td class="tutorial"><b>Tutorials</b></td>
<td class="help-session"><b>Interactive</b></td>
</tr>
</tbody>
</table>

<table>
<thead>

<tr>
<th style="width:5%"></th>
<th style="width:5%">Sunday <br /> 2023/08/13 </th>
<th style="width:18%">Monday <br /> 2023/08/14</th>
<th style="width:18%">Tuesday <br /> 2023/08/15 </th>
<th style="width:18%">Wednesday <br /> 2023/08/16 </th>
<th style="width:18%">Thursday <br /> 2023/08/17 </th>
<th style="width:18%">Friday <br /> 2023/08/18 </th>
</tr>

</thead>

<tbody>

<!-- 8.45 -->
<tr>
<td> 08:45 </td>
<td rowspan="37" class="padding"> Arrival </td>
<td rowspan="2"> Welcome Message<br />
<b>Jeffrey Joy</b> <br />
(<a href="https://www.bccfe.ca/about-us/team/joy-j">BCCfE in HIV/AIDS</a>)</td>
<td> Housekeeping </td>
<td> Housekeeping </td>
<td> Housekeeping </td>
<td> Housekeeping </td>
</tr>

<!------------------------------------------------------------------>
<!-- 9.00-->
<tr>
<td> 09:00 </td>

<!-- Tue -->
<td rowspan="4" class="lecture"><b>Alexei Drummond</b><br />
								<i>Tree priors, clocks and skylines</i><br /><!--(<a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures/raw/master/">slides</a>)-->
</td>

<!-- Wed -->
<td rowspan="4" class="invited-lecture"><b>Ailene MacPherson</b><br />
										(<a href="https://www.sfu.ca/math/people/faculty/ailenem.html">Simon Fraser University</a>)
</td>

<!-- Thu -->
<td rowspan="2" class="invited-lecture"><b>Jūlija Pečerska</b><br />
										(<a href="https://www.zhaw.ch/en/about-us/person/pece/">Zürich University of Applied Sciences</a>)
</td>

<!-- Fri -->
<td rowspan="4" class="invited-lecture"><b>Sungsik (Kevin) Kong</b><br />
										(<a href="https://sungsik-kong.github.io">UW Madison</a>)</td>
</tr>

<!-- 9.15 -->
<tr>
<td> 09:15 </td>

<!-- Mon -->
<td rowspan="3" class="lecture"><b>Alexei Drummond</b><br />
								<i>Introduction to Bayesian phylogenetics</i><br /><!--(<a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures/raw/master/">slides</a>)-->
</td>
</tr>

<!-- 9.30 -->
<tr>
<td> 09:30 </td>
<!-- Thu -->
<td rowspan="2" class="invited-lecture"> <b>David Rasmussen</b><br />(<a href="https://phylodynamics.wordpress.ncsu.edu">NC State University</a>)
</td>
</tr>

<!-- 9.45 -->
<tr>
<td> 09:45 </td>
</tr>

<!------------------------------------------------------------------>
<!-- 10.00 -->
<tr>
<td> 10:00 </td>
<td class="food"><b> Break</b> </td>
<td class="food"><b> Break</b> </td>
<td class="food"><b> Break</b> </td>
<td class="food"><b> Break</b> </td>
<td class="food"><b> Break</b> </td>
</tr>

<!------------------------------------------------------------------>
<!-- 10:15 -->
<tr>
<td> 10:15 </td>

<!-- Mon -->
<td rowspan="3" class="lecture"><b>Louis du Plessis</b><br />
								<i>First steps in BEAST2</i><br /><!--(<a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures/raw/master/">slides</a>)-->
</td>

<!-- Tue -->
<td rowspan="3" class="invited-lecture"> <b>Michael Martin</b><br />(<a href="https://www.iddynamics.jhsph.edu/michael-martin">Johns Hopkins University</a>)</td>

<!-- Wed -->
<td rowspan="3" class="lecture"><b>Joëlle Barido-Sottani</b><br />
								<i>Divergence time estimation</i><br /><!--(<a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures/raw/master/">slides</a>)-->
</td>

<!-- Thu -->
<td rowspan="3" class="lecture"><b>Huw Ogilvie</b><br />
								<i>Species tree inference and the multispecies coalescent</i><br /><!--(<a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures/raw/master/">slides</a>)-->
</td>

<!-- Fri -->
<td rowspan="3" class="lecture"><b>Remco Bouckaert</b><br />
								<i>Model selection and model averaging</i><br /><!--(<a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures/raw/master/">slides</a>)-->
</td>
</tr>

<!-- 10:30 -->
<tr>
<td> 10:30 </td>
</tr>

<!-- 10:45 -->
<tr>
<td> 10:45 </td>
</tr>

<!------------------------------------------------------------------>
<!-- 11:00 -->
<tr>
<td> 11:00 </td>
<td rowspan="2" class="food"><b>Coffee Break</b></td>
<td rowspan="2" class="food"><b>Coffee Break</b></td>
<td rowspan="2" class="food"><b>Coffee Break</b></td>
<td rowspan="2" class="food"><b>Coffee Break</b></td>
<td rowspan="2" class="food"><b>Coffee Break</b></td>
</tr>

<!-- 11:15 -->
<tr>
<td> 11:15 </td>
</tr>


<!------------------------------------------------------------------>
<!-- 11:30 -->
<tr>
<td> 11:30 </td>

<!-- Mon -->
<td rowspan="4" class="tutorial"><b>Introduction to BEAST2</b><br />
								(<a href="/tutorials/Introduction-to-BEAST2/">tutorial</a>)</td>

<!-- Tue -->
<td rowspan="4" class="tutorial"><b>Inference of past population dynamics using skyline models</b><br />
								(<a href="/tutorials/Skyline-plots/">BEAST2 tutorial</a>
								,<a href="https://linguaphylo.github.io/tutorials/skyline-plots/">LPhyBEAST tutorial</a>)</td>

<!-- Wed -->
<td rowspan="4" class="tutorial"><b>Dating species divergences with the fossilized birth-death process</b><br />
								(<a href="/tutorials/FBD-tutorial/">tutorial</a>)</td>

<!-- Thu -->
<td rowspan="4" class="tutorial"><b>Estimating species trees using StarBEAST2</b><br />
								(<a href="/tutorials/starbeast2-tutorial/">tutorial</a>)</td>

<!-- Fri -->
<td rowspan="4" class="tutorial"><b>Substitution model averaging</b><br />
								(<a href="/tutorials/Substitution-model-averaging/">tutorial</a>)</td>

</tr>


<!-- 11:45 -->
<tr>
<td> 11:45 </td>
</tr>


<!-- 12:00 -->
<tr>
<td> 12:00 </td>
</tr>


<!-- 12:15 -->
<tr>
<td> 12:15 </td>
</tr>


<!------------------------------------------------------------------>
<!-- 12:30 -->
<tr>
<td> 12:30 </td>
<td rowspan="4" class="food"> <b>Lunch</b> <br /><i>Howe Sound Brewery</i>  </td>
<td rowspan="4" class="food"> <b>Lunch</b> <br /><i>Howe Sound Brewery</i>  </td>
<td rowspan="2" class="food"> <b>Collect packed lunch</b></td>
<td rowspan="4" class="food"> <b>Lunch</b> <br /><i>Howe Sound Brewery</i>  </td>
<td rowspan="4" class="food"> <b>Lunch</b> <br /><i>Howe Sound Brewery</i>  </td>
</tr>

<!-- 12:45 -->
<tr>
<td> 12:45 </td>
</tr>

<!-- 13:00 -->
<tr>
<td> 13:00 </td>
<td rowspan="20" class="padding"> Free afternoon</td>
</tr>

<!-- 13:15 -->
<tr>
<td> 13:15 </td>
</tr>

<!------------------------------------------------------------------>
<!-- 13:30 -->
<tr>
<td> 13:30 </td>

<td rowspan="4" class="invited-lecture"> <b>Caroline Colijn</b><br />
										(<a href="https://www.sfu.ca/math/people/faculty/ccolijn/">Simon Fraser University</a>)</td>
<td rowspan="2" class="lecture"><b>Jūlija Pečerska</b><br />
								<i>Troubleshooting in BEAST2</i><br /><!--(<a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures/raw/master/">slides</a>)--></td>

<td rowspan="4" class="lecture"><b>Joëlle Barido-Sottani</b> &amp;<br /> <b>David Rasmussen</b><br />
								<i>Structured models in phylodynamics</i><br />
								<br /><!--(<a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures/raw/master/">slides</a>)--></td>
<td rowspan="4" class="help-session"><b>Free Q&amp;A session</b><br />(optional)</td>

</tr>

<!-- 13:45 -->
<tr>
<td> 13:45 </td>
</tr>

<!-- 14:00 -->
<tr>
<td> 14:00 </td>

<!-- Tue -->
<td rowspan="2" class="tutorial"><b>BEAST2 initialization</b><br />
								(<a href="/tutorials/Troubleshooting-initialization-issues">tutorial</a>)</td>

</tr>

<!-- 14:15 -->
<tr>
<td> 14:15 </td>
</tr>

<!------------------------------------------------------------------>
<!-- 14:30 -->
<tr>
<td> 14:30 </td>

<td rowspan="2" class="food"><b>Coffee Break</b> </td>
<td rowspan="2" class="tutorial"><b>BEAST2 convergence</b><br />
								(<a href="/tutorials/Troubleshooting-convergence-issues/">tutorial</a>)</td>

<td rowspan="2" class="food"><b>Coffee Break</b> </td>
<td rowspan="1" class="padding">Departure</td>

</tr>

<!-- 14:45 -->
<tr>
<td> 14:45 </td>
</tr>


<!------------------------------------------------------------------>
<!-- 15:00 -->
<tr>
<td> 15:00 </td>

<td rowspan="4" class="tutorial"><b>Prior selection and clock calibration</b><br />
								(<a href="/tutorials/Prior-selection/">tutorial</a>)</td>
<td rowspan="2" class="food"><b>Coffee Break</b> </td>

<td rowspan="4" class="tutorial"><b>Parameter and state inference using the approximate structured coalescent</b><br />
								(<a href="/tutorials/Mascot-Tutorial/">BEAST2 tutorial</a>,
								,<a href="https://linguaphylo.github.io/tutorials/structured-coalescent/">LPhyBEAST tutorial</a>)</td>
</tr>

<!-- 15:15 -->
<tr>
<td> 15:15 </td>
</tr>

<!-- 15:30 -->
<tr>
<td> 15:30 </td>

<!-- Tue -->
<td rowspan="4" class="lecture"><b>Alexei Drummond</b><br />
								<i>LinguaPhylo</i><br /><!--(<a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures/raw/master/">slides</a>)--></td>

</tr>

<!-- 15:45 -->
<tr>
<td> 15:45 </td>
</tr>


<!------------------------------------------------------------------>
<!-- 16:00 -->
<tr>
<td> 16:00 </td>

<!-- Mon -->
<td rowspan="8" class="invited-lecture"><b>Participant introductions and flash talks</b><br /></td>

<!-- Thu -->
<td rowspan="4" class="tutorial"><b>Population structure using the multi-type birth-death model</b><br />
								 (<a href="/tutorials/Structured-birth-death-model/">tutorial</a>)</td>

</tr>

<tr>
<td> 16:15 </td>
</tr>

<tr>
<td> 16:30 </td>

<!-- Tue -->
<td rowspan="4" class="tutorial"><b>Ancestral state reconstruction using discrete phylogeography</b><br />
								(<a href="https://linguaphylo.github.io/tutorials/discrete-phylogeography/">LPhyBEAST tutorial</a>)</td>

</tr>

<tr>
<td> 16:45 </td>
</tr>

<tr>
<td> 17:00 </td>

<!-- Thu -->
<td rowspan="4" class="help-session"><b>BEAST clinic</b><br />(optional)</td>

</tr>

<tr>
<td> 17:15 </td>
</tr>

<tr>
<td> 17:30 </td>

<!-- Tue -->
<td rowspan="2" class="help-session"><b>BEAST Clinic</b><br />(optional)</td>

</tr>

<tr>
<td> 17:45 </td>
</tr>


<!------------------------------------------------------------------>
<!-- 18:00 -->
<tr>
<td> 18:00 </td>

<td rowspan="1" class="food"> <b>Welcome event</b>  <br /><i>A-Frame Brewing &amp; Flipside Burgers</i> </td>
<td rowspan="1" class="food"> <b>Dinner</b>  <br /><i>Howe Sound Brewery</i> </td>
<td rowspan="1" class="food"> <b>Dinner</b>  <br /><i>Howe Sound Brewery</i> </td>
<td rowspan="1" class="food"> <b>Dinner</b>  <br /><i>Beach BBQ</i> </td>
<td rowspan="1" class="food"> <b>Farewell dinner</b>  <br /><i>Free Bird</i> </td>

</tr>


</tbody>
</table>

<h1 id="instructors">Instructors</h1>

<ul>
  <li><a href="https://bjoelle.github.io">Joëlle Barido-Sottani</a> (Institut de Biologie de l’ENS)</li>
  <li><a href="https://www.cs.auckland.ac.nz/~remco/">Remco Bouckaert</a> (University of Auckland)</li>
  <li><a href="https://profiles.auckland.ac.nz/adru001">Alexei Drummond</a> (University of Auckland)</li>
  <li><a href="https://bsse.ethz.ch/cevo/the-group/people/person-detail.lduplessis.html">Louis du Plessis</a> (ETH Zürich)</li>
  <li><a href="https://profiles.rice.edu/faculty/huw-ogilvie">Huw Ogilvie</a> (Rice University)</li>
  <li><a href="https://allisongroup.blogs.auckland.ac.nz/people/">Alex Popinga</a> (University of Auckland)</li>
  <li><a href="https://www.zhaw.ch/en/about-us/person/pece/">Jūlija Pečerska</a> (Zürich University of Applied Sciences)</li>
  <li><a href="https://phylodynamics.wordpress.ncsu.edu">David Rasmussen</a> (North Carolina State University)</li>
</ul>

<h1 id="lecture-slides">Lecture slides</h1>

<p>Slides to lectures are available from <a href="https://github.com/Taming-the-BEAST/Squamish-BEAST-Tamers-2023-Lectures">here</a>.</p>

<h1 id="required-software">Required Software</h1>

<p>Please ensure the following software is installed prior to the start of the workshop.</p>

<ul>
  <li><strong>BEAST 2:</strong> BEAST 2 is a cross-platform program for Bayesian phylogenetic analysis of molecular sequences. We will be using this software to perform analyses on the data brought to the workshop. We will be using <strong>BEAST v2.7</strong> for the workshop <a href="http://beast2.org/">(download)</a>.</li>
  <li><strong>FigTree:</strong> FigTree is a graphical viewer of phylogenetic trees, specifically designed to display summarised and annotated trees produced by BEAST <a href="http://beast.community/figtree">(download)</a>.</li>
  <li><strong>Tracer:</strong> Tracer is a program for analysing the trace files generated by Bayesian MCMC runs. It can be used to check mixing, ESS, parameter correlations, etc. <a href="http://beast.community/tracer">(download)</a>.</li>
  <li><strong>Java 8 or higher:</strong> You may need to install a Java JDK to run FigTree and Tracer <a href="https://adoptium.net/en-GB/">(download)</a>.</li>
  <li><strong>BEAGLE (optional):</strong> <a href="http://beast.community/beagle">BEAGLE</a> is a high-performance library that can perform the core calculations in BEAST2 (and many other phylogenetics packages) much faster. BEAGLE isn’t necessary for the workshop, but it will speed up your analyses <a href="https://github.com/beagle-dev/beagle-lib">(download)</a>.</li>
  <li><strong>R:</strong> R is a free software environment that provides a wide variety of statistical (linear and nonlinear modelling, classical statistical tests, time-series analysis, classification, clustering) and graphical techniques, and is highly extensible <a href="https://www.r-project.org/">(download)</a>.</li>
  <li><strong>RStudio (optional):</strong> RStudio provides a friendly front-end for R <a href="https://posit.co/download/rstudio-desktop/">(download)</a>.</li>
</ul>

<h1 id="pre-workshop-recommended-reading">Pre-workshop recommended reading</h1>

<h3 id="statistics-and-mathematics-primers">Statistics and mathematics primers</h3>

<p>We are aware that some of the workshop participants do not have a mathematical background and may have difficulty understanding some of the theory. The two links below can help to refresh the basic concepts, but are by no means meant to be comprehensive resources.</p>

<ul>
  <li><a href="http://students.brown.edu/seeing-theory/index.html">Seeing Theory</a>: This website gives a very brief, but pretty and easy to understand introduction to many of the fundamental concepts in statistics.</li>
  <li>Ziheng Yang’s <a href="http://abacus.gene.ucl.ac.uk/PPS/PrimerProbabilityStatistics.pdf">statistics primer</a>: Gives a more in-depth introduction to the topic.</li>
</ul>

<h3 id="phylogenetics-and-phylodynamics-books">Phylogenetics and phylodynamics books</h3>

<p>The books below are not all focused on BEAST or even Bayesian phylogenetics. However, each of the books contains at least one background chapter on the theory of molecular evolution, as well as more in-depth chapters on the models implemented in BEAST and BEAST 2. Most of the books also contain at least some discussion of Bayesian inference.</p>

<ul>
  <li><a href="https://www.beast2.org/book/">Bayesian Evolutionary Analysis with BEAST</a>, <em>Alexei J. Drummond, Remco R. Bouckaert</em></li>
  <li><a href="http://abacus.gene.ucl.ac.uk/MESA/">Molecular Evolution: A Statistical Approach</a>, <em>Ziheng Yang</em></li>
  <li><a href="http://www.cambridge.org/catalogue/catalogue.asp?isbn=9780521877107">The Phylogenetic Handbook</a>, <em>Philippe Lemey, Marco Salemi, Anne-Mieke Vandamme</em></li>
  <li><a href="http://eu.wiley.com/WileyCDA/WileyTitle/productCd-0865428891.html">Molecular Evolution: A Phylogenetic Approach</a>, <em>Roderick D.M. Page, Edward C. Holmes</em></li>
  <li><a href="https://www.amazon.co.uk/Inferring-Phylogenies-Joseph-Felsenstein/dp/0878931775">Inferring Phylogenies</a>, <em>Joseph Felsenstein</em></li>
</ul>

<p>In addition, the following review papers might be useful:</p>

<ul>
  <li><a href="https://www.nature.com/articles/nrg.2015.8">Bayesian molecular clock dating of species divergences in the genomics era</a>, <em>Mario dos Reis, Phillip Donoghue, Ziheng Yang</em></li>
  <li><a href="https://www.nature.com/articles/nrg2583">Evolutionary analysis of the dynamics of viral infectious disease</a>, <em>Oliver Pybus, Andrew Rambaut</em></li>
  <li><a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/brv.12390">Bayesian molecular dating: opening up the black box</a>, <em>Lindell Bromham, Sebastián Duchêne, Xia Hua, Andrew M. Ritchie, David A. Duchêne, Simon Y. W. Ho</em></li>
</ul>

	</div>
</div>

	
	</div>

	<div class="footerspacer"></div>

<div class="footer">
  <div class="container">
    <div class="row">
      <div class="col-md-6">
        <ul class="list-group list-group-horizontal">
        <li class="footernav list-group-item"><a class="foot" href="/about/">about</a></li>
        <li class="footernav list-group-item"><a class="foot" href="/contact/">contact</a></li>
        <li class="footernav list-group-item"><a class="foot" href="/license/">license</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

	
	<!--div id="footer"><span style="display:none">foo</span></div-->
		
</body>
</html>

