<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="utf-8"/>
	<title>MASCOT v1.2.2 Tutorial</title>
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	
	<!-- RSS feed -->
	<link rel="alternate" type="application/rss+xml" title="Taming the BEAST" href="http://localhost:4000/feed.news.xml">
	<link rel="alternate" type="application/rss+xml" title="Taming the BEAST" href="http://localhost:4000/feed.tutorials.xml">

	
    <!-- Customized Bootstrap + Font Awesome + Solarized -->
    <link href="/css/style.css" rel="stylesheet" media="screen">

	<!-- Favicon -->
	<link rel="shortcut icon" href="/images/favicon.ico"/>	

	<!-- Typekit -->
	<script>
	  (function(d) {
		var config = {
		  kitId: 'xeu8jut',
		  scriptTimeout: 3000
		},
		h=d.documentElement,t=setTimeout(function(){h.className=h.className.replace(/\bwf-loading\b/g,"")+" wf-inactive";},config.scriptTimeout),tk=d.createElement("script"),f=false,s=d.getElementsByTagName("script")[0],a;h.className+=" wf-loading";tk.src='//use.typekit.net/'+config.kitId+'.js';tk.async=true;tk.onload=tk.onreadystatechange=function(){a=this.readyState;if(f||a&&a!="complete"&&a!="loaded")return;f=true;clearTimeout(t);try{Typekit.load(config)}catch(e){}};s.parentNode.insertBefore(tk,s)
	  })(document);
	</script>
		
	<!-- Google Analytics -->	
	<script>
	  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

      ga('create', 'UA-86786826-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');

	</script>
	
	<!-- jQuery -->
	<script src="/js/jquery.min.js"></script>
	
	<!-- Bootstrap -->
	<script src="/js/transition.js"></script>
	<script src="/js/collapse.js"></script>


</head>


<body>
	
	<div id="header">
	<nav class="navbar navbar-default navbar-static-top" role="navigation">	 
		<div class="container">	

			<div class="navbar-header">
				<button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar-collapse-element" aria-expanded="false">
					<span class="sr-only">Toggle navigation</span>
					<span class="icon-bar"></span>
					<span class="icon-bar"></span>
					<span class="icon-bar"></span>
				</button>
									
				<a class="navbar-brand" href="/">Taming the BEAST</a>
				<object class="logo navbar-brand" data="/images/logo_bw.svg" type="image/svg+xml"></object>
				
			</div>

			<div class="collapse navbar-collapse" id="navbar-collapse-element">
				<ul class="nav navbar-nav navbar-right">					
					
					<li>
					
					<a href="/news/">news</a></li>

					
                                        <li>
					
                                        <a href="/workshops/">workshops</a></li>

					
					<li>
					
					<a href="/tutorials/">tutorials</a></li>	    	  

					
					<li>
					
					<a href="/contribute/">contribute</a></li>
					<li><a type="application/rss+xml" href="/feed.news.xml"><i class="fa fa-rss"></i></a></li>
				</ul>
			</div>
			
		</div>
	</nav>	
</div>

<!--div class="headerspacer"></div-->

	
	<div class="container">	
		
	<script src="/js/sidebar.js"></script>




<div class="row">
	<div class="hidden-xs hidden-sm col-md-12">
		<div class="bigtitle titlebox">
			<ol class="breadcrumb">
			
			
				
				
				
				<li><a class="off" href="/tutorials/Mascot-Tutorial/">MASCOT v1.2.2 Tutorial</a></li>
				
			
			</ol>
		</div>
		<p>
		<div class="head">
			
				Parameter and State inference using the approximate structured coalescent
			
		</div>
		
		<div class="smallhead">
			by
			
			
				
						Nicola F. MÃ¼ller
					
						
					
			
		</div>
		
	</div>				
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-3">
	<div id="sidebar">
		<div class="bigspacer"></div>
		<div class="smallhead">
			Tutorial
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-file-text-o fa-fw"></i>
					<a class="off" href="/tutorials/Mascot-Tutorial/Mascot-Tutorial.pdf">Mascot-Tutorial.pdf</a>	
				</li>
			
			<div class="smallspacer"></div>
			<li>
				<i class="fa fa-github fa-fw"></i>
				<a class="off" href="https://github.com/nicfel/Mascot-Tutorial" target="_blank">Github repository</a>
			</li>
			<div class="smallspacer"></div>
			<li>
				<i class="fa fa-certificate fa-fw"></i>
				<a class="off" href="LICENSE.html">
				License</a>
			</li>
			<div class="smallspacer"></div>
			<li>			
				<i class="fa fa-bar-chart fa-fw"></i>
				<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/graphs/traffic" target="_blank">
				Statistics</a>
			</li>
			</ul>
		</div>		
		<div class="spacer"></div>
		 	
		<div class="smallhead">
			Data
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-align-justify fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/raw/master/data/H3N2.nexus" target="_blank" rel="nofollow">
						H3N2.nexus
					</a>
				</li>
			
			</ul>
		</div>		
		<div class="spacer"></div>
		
		 	
		<div class="smallhead">
			XML
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-code fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/raw/master/xml/H3N2.xml" target="_blank" download rel="nofollow">
						H3N2.xml
					</a>
				</li>
			
			</ul>
		</div>	
		<div class="spacer"></div>
		
		
		 
		<div class="smallhead">
			Output
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-file-text-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/raw/master/precooked_runs/H3N2.log" target="_blank" download rel="nofollow">
						H3N2.log
					</a>
				</li>
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-file-text-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/raw/master/precooked_runs/H3N2.trees" target="_blank" download rel="nofollow">
						H3N2.trees
					</a>
				</li>
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-file-text-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/raw/master/precooked_runs/H3N2.mcc.trees" target="_blank" download rel="nofollow">
						H3N2.mcc.trees
					</a>
				</li>
			
			</ul>
		</div>		
		
		<div class="spacer"></div>
		<div class="smallhead">
			Contributors
		</div>	
		<div class="pad-left note">
				
			<div class="smallspacer"></div>				
			<div>
				<a class="off" href="https://github.com/laduplessis">
    			<img class="pull-left avatar" src="https://avatars3.githubusercontent.com/u/8872277?v=4&s=50">
    			<div class="handlebox" style="padding-left:5px;">
    				laduplessis
    			</div>
 				</a>
 			</div>			
 				
			<div class="smallspacer"></div>				
			<div>
				<a class="off" href="https://github.com/tgvaughan">
    			<img class="pull-left avatar" src="https://avatars1.githubusercontent.com/u/512538?v=4&s=50">
    			<div class="handlebox" style="padding-left:5px;">
    				tgvaughan
    			</div>
 				</a>
 			</div>			
 				
			<div class="smallspacer"></div>				
			<div>
				<a class="off" href="https://github.com/nicfel">
    			<img class="pull-left avatar" src="https://avatars0.githubusercontent.com/u/19623728?v=4&s=50">
    			<div class="handlebox" style="padding-left:5px;">
    				nicfel
    			</div>
 				</a>
 			</div>			
 			
		</div>
		<div class="bigspacer"></div>
		<div class="smallhead">
			Latest commits
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/commit/390e9876fa01dbf85427a5f6f6673cb98e659d50">
					11 Oct 2019 - <span class="text-gray">adds error section</span>
					</a> 
				</li>
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/commit/ecb3c9d5f2f2b1436b88ca80780939dd3a7710d4">
					20 Feb 2019 - <span class="text-gray">updates version</span>
					</a> 
				</li>
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/commit/e1fa209a4464709cedb22d21673740934d8b694b">
					19 Jun 2018 - <span class="text-gray">changed clock rate prior</span>
					</a> 
				</li>
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/commit/83de382357b47d466921c21094bac14266538fbf">
					19 Jun 2018 - <span class="text-gray">added link to Peter Beerlis backwards migration explanation</span>
					</a> 
				</li>
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Mascot-Tutorial/commit/1a9f71e8227161f3e35b3b6dcae1119a7b29a0c8">
					18 Jun 2018 - <span class="text-gray">citation and version update</span>
					</a> 
				</li>
			
			</ul>
		</div>		
		<div class="spacer"></div>	
	</div>
	</div>

	<div class="col-md-8">				
		<div class="post">
			<p>
			<h1>Background</h1>

<p>Phylogeographic methods can help reveal the movement of genes between populations of organisms. This has been widely used to quantify pathogen movement between different host populations, the migration history of humans, and the geographic spread of languages or the gene flow between species using the location or state of samples alongside sequence data. Phylogenies therefore offer insights into migration processes not available from classic epidemiological or occurrence data alone.</p>

<p>The structured coalescent allows to coherently model the migration and coalescent process, but struggles with complex datasets due to the need to infer ancestral migration histories. Thus, approximations to the structured coalescent, which integrate over all ancestral migration histories, have been developed. This tutorial gives an introduction into how a MASCOT analysis in BEAST2 can be set-up. MASCOT is short for <strong>M</strong>arginal <strong>A</strong>pproximation of the <strong>S</strong>tructured <strong>CO</strong>alescen <strong>T</strong>  <a class="citation" href="#mueller2017mascot">(MÃ¼ller, Rasmussen, &amp; Stadler, 2018)</a> and implements a structured coalescent approximation <a class="citation" href="#Mueller2017">(MÃ¼ller, Rasmussen, &amp; Stadler, 2017)</a>. This approximation doesn&#39;t require migration histories to be sampled using MCMC and therefore allows to analyse phylogenies with more than three or four states.</p>

<hr>

<h1>Programs used in this Exercise</h1>

<h3>BEAST2 - Bayesian Evolutionary Analysis Sampling Trees 2</h3>

<p>BEAST2 (<a href="http://www.beast2.org">http://www.beast2.org</a>) is a free software package for Bayesian evolutionary analysis of molecular sequences using MCMC and strictly oriented toward inference using rooted, time-measured phylogenetic trees. This tutorial is written for BEAST v2.5 <a class="citation" href="#BEAST2book2014">(Drummond &amp; Bouckaert, 2014)</a>.</p>

<h3>BEAUti2 - Bayesian Evolutionary Analysis Utility</h3>

<p>BEAUti2 is a graphical user interface tool for generating BEAST2 XML configuration files.</p>

<p>Both BEAST2 and BEAUti2 are Java programs, which means that the exact same code runs on all platforms. For us it simply means that the interface will be the same on all platforms. The screenshots used in this tutorial are taken on a Mac OS X computer; however, both programs will have the same layout and functionality on both Windows and Linux. BEAUti2 is provided as a part of the BEAST2 package so you do not need to install it separately.</p>

<h3>TreeAnnotator</h3>

<p>TreeAnnotator is used to summarise the posterior sample of trees to produce a maximum clade credibility tree. It can also be used to summarise and visualise the posterior estimates of other tree parameters (e.g. node height).</p>

<p>TreeAnnotator is provided as a part of the BEAST2 package so you do not need to install it separately.</p>

<h3>Tracer</h3>

<p>Tracer (<a href="http://tree.bio.ed.ac.uk/software/tracer">http://tree.bio.ed.ac.uk/software/tracer</a>) is used to summarise the posterior estimates of the various parameters sampled by the Markov Chain. This program can be used for visual inspection and to assess convergence. It helps to quickly view median estimates and 95% highest posterior density intervals of the parameters, and calculates the effective sample sizes (ESS) of parameters. It can also be used to investigate potential parameter correlations. We will be using Tracer v1.7.0.</p>

<h3>FigTree</h3>

<p>FigTree (<a href="http://tree.bio.ed.ac.uk/software/figtree">http://tree.bio.ed.ac.uk/software/figtree</a>) is a program for viewing trees and producing publication-quality figures. It can interpret the node-annotations created on the summary trees by TreeAnnotator, allowing the user to display node-based statistics (e.g. posterior probabilities). We will be using FigTree v1.4.2.</p>

<hr>

<h1>Practical: Parameter and State inference using the approximate structured coalescent</h1>

<p>In this tutorial we will estimate migration rates, effective population sizes and locations of internal nodes using the marginal approximation of the structured coalescent implemented in BEAST2, MASCOT <a class="citation" href="#mueller2017mascot">(MÃ¼ller, Rasmussen, &amp; Stadler, 2018)</a>.</p>

<p>The aim is to:</p>

<ul>
<li> Learn how to infer structure from trees with sampling location</li>
<li> Get to know how to choose the set-up of such an analysis</li>
<li> Learn how to read the output of a MASCOT analysis</li>
</ul>

<h2>Setting up an analysis in BEAUti</h2>

<h3>Download MASCOT</h3>

<p>First, we have to download the package MASCOT using the BEAUTi package manager. Go to <em>File &gt;&gt; Manage Packages</em> and download the package MASCOT.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:50%;" src="figures/MascotDownload.png" alt="">
    <figcaption>Figure 1: Download the MASCOT package.</figcaption>
</figure>

<p>MASCOT will only be available in BEAUti once you close and restart the program.</p>

<h3>Loading the template</h3>

<p>Next, we have to load the BEAUTi template from <em>File</em>, select <em>Template &gt;&gt; Mascot</em>.</p>

<h3>Loading the Influenza A/H3N2 Sequences (Partitions)</h3>

<p>The sequence alignment is in the file <a href="http://github.com/nicfel/Mascot-Tutorial/raw/master/data/H3N2.nexus">H3N2.nexus</a>. Right-click on this link and save it to a folder on your computer. Once downloaded, this file can either be drag-and-dropped into BEAUti or added by using BEAUti&#39;s menu system via <em>File &gt;&gt; Import Alignment</em>. Once the sequences are added, we need to specify the sampling dates and locations.</p>

<h3>Get the sampling times (Tip Dates)</h3>

<p>Open the &quot;Tip Dates&quot; panel and then select the &quot;Use tip dates&quot; checkbox.</p>

<p>The sampling times are encoded in the sequence names.  We can tell BEAUti to use these by clicking the <em>Auto-configure</em> button. The sampling times appear following the third vertical bar &quot;|&quot; in the sequence name. To extract these times, select &quot;split on character&quot;, enter &quot;|&quot; (without the quotes) in the text box immediately to the right, and then select &quot;3&quot; from the drob-down box to the right, as shown in the figure below.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/TipDates.png" alt="">
    <figcaption>Figure 2: Guess sampling times.</figcaption>
</figure>

<p>Clicking &quot;Ok&quot; should now populate the table with the sample times extracted from the sequence names: the column <strong>Date</strong> should now have values between 2000 and 2002 and the column <strong>Height</strong> should have values from 0 to 2. The heights denote the time difference from a sequence to the most recently sampled sequence. If everything is specified correctly, the sequence with Height 0.0 should have Date 2001.9. Next, the sampling locations need to be specified.</p>

<h3>Get the sampling locations (Tip Locations)</h3>

<p>As for the sampling times, sampling locations can be extracted from
the sequence names.  Select the &quot;Tip Locations&quot; panel.  Initially the
column <strong>Location</strong> should be NOT_SET for every sequence.  After
clicking the <em>Guess</em> button, you can split the sequence on the
vertical bar &quot;|&quot; again by selecting &quot;split on character&quot; and entering
&quot;|&quot; in the box. However, the locations are in the fourth group, so
this time choose &quot;4&quot; from the drop-down menu. After clicking the <em>OK</em>
button, the window should look like the one shown in the figure below:</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/TipLocations.png" alt="">
    <figcaption>Figure 3: Configuring sample locations.</figcaption>
</figure>

<h3>Specify the Site Model (Site Model)</h3>

<p>Next, we have to specify the site model. To do this, choose the &quot;Site Model&quot; tab. For Influenza Hemagluttanin sequences as we have here, HKY is the most commonly used model of nucleotide evolution. This model allows for differences in transversion and transition rates, meaning that changes between bases that are chemically more closely related (transitions) are allowed to have a different rate to changes between bases that chemically more distinct (transversions). Additionally, we should allow for different rate categories for different sires in the alignment. This can be done by setting the <em>Gamma Category Count</em> to 4, which is just a value that has typically been used. Make sure that estimate is checked next to the shape parameter. To reduce the number of parameters we have to estimate, we can set Frequencies to Empirical.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/SiteModel.png" alt="">
    <figcaption>Figure 4: Set the site model.</figcaption>
</figure>

<h3>Set the clock model (Clock Model)</h3>

<p>For rapidly evolving viruses, the assumption of a strict molecular clock is often made, meaning that the molecular clock is the same on each branch of the phylogeny. To decrease the burnin phase, we can set the initial value to 0.005.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/ClockRate.png" alt="">
    <figcaption>Figure 5: Set the initial clock rate.</figcaption>
</figure>

<h3>Specify the priors (Priors)</h3>

<p>Now, we need to set the priors for the various parameters of the model. We do this by switching to the &quot;Priors&quot; tab.</p>

<p>First, consider the effective population size parameter.  Since we have only a few samples per location, meaning little information about the different effective population sizes, we will need an informative prior. In this case we will use a log normal prior with parameters M=0 and S=1.  (These are respectively the mean and variance of the corresponding normal distribution in log space.)  To use this prior, choose &quot;Log Normal&quot; from the dropdown menu to the right of the Ne.t:H3N2 parameter label, then click the arrow to the left of the same label and fill in the parameter values appropriately (i.e. M=0 and S=1). Ensure that the &quot;mean in real space&quot; checkbox remains unchecked.</p>

<p>The existing exponential distribution as a prior on the migration rate puts much weight on lower values while not prohibiting larger ones. For migration rates, a prior that prohibits too large values while not greatly distinguishing between very small and very <em>very</em> small values is generally a good choice. Be aware however that the exponential distirbution is quite an informative prior: one should be careful that to choose a mean so that feasible rates are at least within the 95% HPD interval of the prior.  (This can be determined by clicking the arrow to the left of the parameter name and looking at the values below the graph that appears on the right.)</p>

<p>Finally, set the prior for the clock rate. We have a good idea about the clock rate of Influenza A/H3N2 Hemagglutinin. From previous work by other people, we know that the clock rate will be around 0.005 substitution per site per year. To include that prior knowledger, we can set the prior on the clock rate to a log normal distribution with mean in <strong>real space</strong>. To specify the mean in real space, make sure that the box <em>Mean In Real Space</em> is checked. If we set the S value to 0.25, we say that we expect the clock rate to be with 95% certainty between 0.00321 and 0.00731.
<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/Priors.png" alt="">
    <figcaption>Figure 6: Set up of the prior distributions.</figcaption>
</figure></p>

<h3>Specify the MCMC chain length (MCMC)</h3>

<p>Now switch to the &quot;MCMC&quot; tab. Here we can set the length of the MCMC
chain and decide how frequently the parameter and trees are
logged. For this dataset, 2 million iterations should be
sufficient. In order to have enough samples but not create too large
files, we can set the logEvery to 5000, so we have 401 samples
overall. Next, we have to save the <code>*.xml</code> file using <em>File &gt;&gt; Save
as</em>.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/MCMC.png" alt="">
    <figcaption>Figure 7: save the *.xml.</figcaption>
</figure>

<h3>Run the Analysis using BEAST2</h3>

<p>Run the <code>*.xml</code> using BEAST2 or use finished runs from the <em>precooked-runs</em> folder. The analysis should take about 6 to 7 minutes. If you want to learn some more about what the migration rates we actually estimate, have a look at this blog post of Peter Beerli <a href="http://popgen.sc.fsu.edu/Migrate/Blog/Entries/2013/3/22_forward-backward_migration_rates.html">http://popgen.sc.fsu.edu/Migrate/Blog/Entries/2013/3/22_forward-backward_migration_rates.html</a>.</p>

<h3>Analyse the log file using Tracer</h3>

<p>First, we can open the <code>*.log</code> file in tracer to check if the MCMC has converged. The ESS value should be above 200 for almost all values and especially for the posterior estimates.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/LogPosterior.png" alt="">
    <figcaption>Figure 8: Check if the posterior converged.</figcaption>
</figure>

<p>We can have a look at the marginal posterior distributions for the effective population sizes. New York is inferred to have the largest effective population size before Hong Kong and New Zealand. This tells us that two lineages that are in New Zealand are expected to coalesce quicker than two lineages in Hong Kong or New York.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/LogNe.png" alt="">
    <figcaption>Figure 9: Compare the different inferred effective population sizes.</figcaption>
</figure>

<p>In this example, we have relatively little information about the effective population sizes of each location. This can lead to estimates that are greatly informed by the prior. Additionally, there can be great differences between median and mean estimates. The median estimates are generally more reliable since they are less influence by extreme values.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/MeanMedian.png" alt="">
    <figcaption>Figure 10: Differences between Mean and Meadian estimates.</figcaption>
</figure>

<p>We can then look at the inferred migration rates. The migration rates have the label b_migration.*, meaning that they are backwards in time migration rates. The highest rates are from New York to Hong Kong. Because they are backwards in time migration rates, this means that lineages from New York are inferred to be likely from Hong Kong if we&#39;re going backwards in time. In the inferred phylogenies, we should therefore make the observation that lineages ancestral to samples from New York are inferred to be from Hong Kong backwards.</p>

<p>A more in depth explanation of what backwards migration really are can be found here <a href="http://popgen.sc.fsu.edu/Migrate/Blog/Entries/2013/3/22_forward-backward_migration_rates.html">http://popgen.sc.fsu.edu/Migrate/Blog/Entries/2013/3/22_forward-backward_migration_rates.html</a></p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/LogMigration.png" alt="">
    <figcaption>Figure 11: Compare the inferrred migration rates.</figcaption>
</figure>

<h3>Make the MCC tree using TreeAnnotator</h3>

<p>Next, we want to summarize the trees. This we can do using TreeAnnotator. Open the program and then set the options as below. You have to specify the <em>Burnin percentage</em>, the <em>Node heights</em>, <em>Input Tree File</em> and the <em>Output File</em>. After clicking <em>Run</em> the program should summarize the trees.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:50%;" src="figures/TreeAnnotator.png" alt="">
    <figcaption>Figure 12: Make the maximum clade credibility tree.</figcaption>
</figure>

<h3>Check the MCC tree using FigTree</h3>

<p>In each logging step of the tree during the MCMC, MASCOT logs several different things. It logs the inferred probability of each node being in any possible location. In this example, these would be the inferred probabilities of being in Hong Kong, New York and New Zealand. Additonally, it logs the most likely location of each node.</p>

<p>After opening the MCC tree in FigTree, we can visualize several things.
To color branches, you can go to <em>Appearance &gt;&gt; Colour by</em> and select <em>max</em>. This is the location that was inferred to be most often the most likely location of the node.</p>

<figure>
<a id="fig:example1"></a>
<img style="width:100%;" src="figures/ColorsTree.png" alt="">
<figcaption>Figure 13: Inferred node locations.</figcaption>
</figure>

<p>We can now determine if lineages ancestral to samples from New York are actually inferred to be from Hong Kong, or the probability of the root being in any of the locations.</p>

<p>To get the actual inferred probabilities of each node being in any of the 3 locations, you can go to <em>Node Labels &gt;&gt; Display</em> an then choose Hong_Kong, New_York or New_Zealand. These are the actual inferred probabilities of the nodes being in any location.</p>

<p>It should however be mentioned that the inference of nodes being in a particular location makes some simplifying assumptions, such as that there are no other locations (i.e. apart from the sampled locations) where lineages could have been.</p>

<p>Another important thing to know is that currently, we assume rates to be constant. This means that we assume that the population size of the different locations does not change over time. We also make the same assumption about the migration rates through time.</p>

<h3>Errors that can occur (Work in progress)</h3>

<p>One of the errors message that can occur regularly is the following:
<code>too many iterations, return negative infinity</code>
This occurs when the integration step size of the ODE&#39;s to compute the probability of observing a phylogenetic tree in MASCOT is becoming too small.
This generally occurs if at least one migration rate is really large or at least one effective population size is really small (i.e. the coalescent rate is really high).
This causes integration steps to be extremely small, which in turn would require a lot of time to compute the probability of a phylogenetic tree under MASCOT.
Instead of doing that, this state is rejected by assigning its log probability the value negative infinity.</p>

<p>This error can have different origins and a likely incomplete list is the following:
1. The priors on migration rates put too much weight on really high rates. To fix this, reconsider your priors on the migration rates. Particularly, check if the prior on the migration rates make sense in comparison to the height of the tree. If, for example, the tree has a height of 1000 years, but the prior on the migration rate is exponential with mean 1, then the prior assumption is that between any two states, we expected approximately 1000 migration events.
2. The prior on the effective population sizes is too low, meaning that the prior on the coalescent rates (1 over the effective population size) is too high. This can for example occur when the prior on the effective population size was chosen to be 1/X. To fix, reconsider your prior on the effective population size.
3. There is substantial changes of the effective population sizes and/or migration rates over time that are not modeled. In that case, changes in the effective population sizes or migration rates have to be explained by population structure, which can again lead to some effective population sizes being very low and some migration rates being very high. In that case, there is unfortunately not much that can be done, since MASCOT is not an appropriate model for the dataset.
4. There is strong subpopulation structure within the different subpopulations used. In that case, reconsider if the individual sub-populations used are reasonable.</p>

<hr>

<h1>Useful Links</h1>

<p>If you interested in the derivations of the marginal approximation of the structured coalescent, you can find them here <a class="citation" href="#Mueller2017">(MÃ¼ller, Rasmussen, &amp; Stadler, 2017)</a>. This paper also explains the mathematical differences to other methods such as the theory underlying BASTA. To get a better idea of how the states of internal nodes are calculated, have a look in this paper <a class="citation" href="#mueller2017mascot">(MÃ¼ller, Rasmussen, &amp; Stadler, 2018)</a>.</p>

<ul>
<li>MASCOT source code: <a href="https://github.com/nicfel/Mascot">https://github.com/nicfel/Mascot</a></li>
<li><a href="http://www.beast2.org/book.html">Bayesian Evolutionary Analysis with BEAST 2</a> <a class="citation" href="#BEAST2book2014">(Drummond &amp; Bouckaert, 2014)</a></li>
<li>BEAST 2 website and documentation: <a href="http://www.beast2.org/">http://www.beast2.org/</a></li>
<li>Join the BEAST user discussion: <a href="http://groups.google.com/group/beast-users">http://groups.google.com/group/beast-users</a></li>
</ul>

<hr>

<h1>Relevant References</h1>

<ol class="bibliography"><li><span id="mueller2017mascot">MÃ¼ller, N. F., Rasmussen, D., &amp; Stadler, T. (2018). MASCOT: Parameter and state inference under the marginal structured coalescent approximation. <i>Bioinformatics</i>, bty406. https://doi.org/10.1093/bioinformatics/bty406</span></li>
<li><span id="Mueller2017">MÃ¼ller, N. F., Rasmussen, D. A., &amp; Stadler, T. (2017). The Structured Coalescent and its Approximations. <i>Molecular Biology and Evolution</i>, msx186.</span></li>
<li><span id="BEAST2book2014">Drummond, A. J., &amp; Bouckaert, R. R. (2014). <i>Bayesian evolutionary analysis with BEAST 2</i>. Cambridge University Press.</span></li></ol>


			<hr>
			<h1>Citation</h1>

			
<p>If you found <strong>Taming the BEAST</strong> helpful in designing your research, please cite the following paper:</p>

<p style="margin-left: 40px">
JoÃ«lle Barido-Sottani, Veronika BoÅ¡kovÃ¡, Louis du Plessis, Denise KÃ¼hnert, Carsten Magnus, Venelin Mitov, Nicola F. MÃ¼ller, JÅ«lija PeÄerska, David A. Rasmussen, Chi Zhang, Alexei J. Drummond, Tracy A. Heath, Oliver G. Pybus, Timothy G. Vaughan, Tanja Stadler (2018). Taming the BEAST â A community teaching material resource for BEAST 2. <em>Systematic Biology</em>, <em>67(1)</em>, 170â-174. doi: <a target="_blank" href="https://doi.org/10.1093/sysbio/syx060">10.1093/sysbio/syx060</a>
</p> 
		</div>
	</div>			
	<div class="col-md-1"></div>
</div>


	
	</div>

	<div class="footerspacer"></div>

<div class="footer">
	<div class="container">
		<div class="col-md-6 reduced-gutter">
			<div class="bigspacer"></div>		
				<ul class="list-inline">
					<li class="footernav">
						<i class="fa"></i> <a class="foot" href="/about/">about</a>
					</li>	
					<li class="footernav">
						<i class="fa"></i> <a class="foot" href="/contact/">contact</a>
					</li>	
					<li class="footernav">
						<i class="fa"></i> <a class="foot" href="/license/">license</a>
					</li>	
				</ul>
			<div class="spacer"></div>			
		</div>
		
		<div class="col-md-5 text-right">
			<div class="bigspacer"></div>		
				<ul class="list-inline">					
					<li class="footernav">
						<i class="fa"></i>powered by <a class="foot" href="http://pages.github.com">GitHub</a> and <a class="foot" href="http://www.jekyllrb.com">Jekyll</a>
					</li>								
				</ul>
			<div class="spacer"></div>			
		</div>

		<div class="col-md-1 reduced-gutter">
			<div class="spacer"></div>						
				<div class="title text-gray">
					<a class="foot" href="http://www.github.com/taming-the-beast"><i class="fa fa-github fa-fw"></i></a>
				</div>
			<div class="spacer"></div>			
		</div>
	</div>
</div>

	
	<!--div id="footer"><span style="display:none">foo</span></div-->
		
</body>
</html>

