<p>% This file was created with JabRef 2.9.2.
% Encoding: MacRoman</p>

<p>Bouckaert, R., Heled, J., Kühnert, D., Vaughan, T., Wu, C.-H., Xie, D., Suchard, M. A., Rambaut, A., &amp; Drummond, A. J. (2014). BEAST 2: a software platform for Bayesian evolutionary analysis. <i>PLoS Computational Biology</i>, <i>10</i>(4), e1003537. https://doi.org/10.1371/journal.pcbi.1003537</p>

<p>Drummond, A. J., &amp; Bouckaert, R. R. (2014). <i>Bayesian evolutionary analysis with BEAST 2</i>. Cambridge University Press.</p>

<p>Bouckaert, R., Vaughan, T. G., Barido-Sottani, J., Duchêne, S., Fourment, M., Gavryushkina, A., Heled, J., Jones, G., Kühnert, D., Maio, N. D., Matschiner, M., Mendes, F. K., Müller, N. F., Ogilvie, H. A., Plessis, L. du, Popinga, A., Rambaut, A., Rasmussen, D., Siveroni, I., … Drummond, A. J. (2019). BEAST 2.5: An advanced software platform for Bayesian evolutionary analysis. <i>PLOS Computational Biology</i>, <i>15</i>(4).</p>

<p>Heath, T. A., Huelsenbeck, J. P., &amp; Stadler, T. (2014). The fossilized birth-death process for coherent calibration of divergence-time estimates. <i>Pnas</i>, <i>111</i>, E2957–E2966.</p>

<p>Team, R. C. (2022). <i>R: A Language and Environment for Statistical Computing</i>. R Foundation for Statistical Computing. https://www.R-project.org/</p>

<p>Wickham, H. (2016). <i>ggplot2: Elegant Graphics for Data Analysis</i>. Springer-Verlag New York. https://ggplot2.tidyverse.org</p>

<p>Yu, G., Smith, D., Zhu, H., Guan, Y., &amp; Lam, T. T.-Y. (2017). ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. <i>Methods in Ecology and Evolution</i>, <i>8</i>(1), 28–36. https://doi.org/10.1111/2041-210X.12628</p>

<p>Revell, L. J. (2012). phytools: An R package for phylogenetic comparative biology (and other things). <i>Methods in Ecology and Evolution</i>, <i>3</i>, 217–223. https://doi.org/10.1111/j.2041-210X.2011.00169.x</p>

<p>Paradis, E., &amp; Schliep, K. (2019). ape 5.0: an environment for modern phylogenetics and evolutionary analyses in R. <i>Bioinformatics</i>, <i>35</i>, 526–528. https://doi.org/10.1093/bioinformatics/bty633</p>

<p>Vaughan, T. G. (2017). IcyTree: rapid browser-based visualization for phylogenetic trees and networks. <i>Bioinformatics</i>, <i>33</i>(15), 2392–2394.</p>

<p>Xiao, N. (2024). <i>ggsci: Scientific Journal and Sci-Fi Themed Color Palettes for ’ggplot2.’</i> https://CRAN.R-project.org/package=ggsci</p>

<p>Wang, L.-G., Lam, T. T.-Y., Xu, S., Dai, Z., Zhou, L., Feng, T., Guo, P., Dunn, C. W., Jones, B. R., Bradley, T., Zhu, H., Guan, Y., Jiang, Y., &amp; Yu, G. (2020). treeio: an R package for phylogenetic tree input and output with richly annotated and associated data. <i>Molecular Biology and Evolution</i>, <i>37</i>(2), 599–603. https://doi.org/10.1093/molbev/msz240</p>

<p>Yu, G. (2022). <i>Data Integration, Manipulation and Visualization of Phylogenetic Treess</i> (1st edition). Chapman and Hall/CRC. https://doi.org/10.1201/9781003279242</p>

<p>du Plessis, L. (2023). <i>beastio: Functions for pre- and post-processing of BEAST and BEAST2 files.</i></p>

<p>M&lt;U+00FC&gt;ller, K., &amp; Wickham, H. (2023). <i>tibble: Simple Data Frames</i>. https://CRAN.R-project.org/package=tibble</p>

<p>Wickham, H., Vaughan, D., &amp; Girlich, M. (2023). <i>tidyr: Tidy Messy Data</i>. https://CRAN.R-project.org/package=tidyr</p>

<p>Wilke, C. O. (2019). <i>Fundamentals of data visualization: a primer on making informative and compelling figures</i>. O’Reilly Media.</p>

<p>Chang, W. (2018). <i>R graphics cookbook: practical recipes for visualizing data</i>. O’Reilly Media.</p>

<p>du Plessis, L. (2016). <i>bdskytools: BDSKY Tools</i>.</p>

<p>Yu, G. (2025). <i>treedataverse: Easily Install and Load the ’treedataverse.’</i></p>
