<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="utf-8"/>
	<title>Estimating Reassortment Networks with CoalRe</title>
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	
	<!-- RSS feed -->
	<link rel="alternate" type="application/rss+xml" title="Taming the BEAST" href="http://localhost:4000/feed.news.xml">
	<link rel="alternate" type="application/rss+xml" title="Taming the BEAST" href="http://localhost:4000/feed.tutorials.xml">

	
    <!-- Customized Bootstrap + Font Awesome + Solarized -->
    <link href="/css/style.css" rel="stylesheet" media="screen">

	<!-- Favicon -->
	<link rel="shortcut icon" href="/images/favicon.ico"/>	

	<!-- Typekit -->
	<script>
	  (function(d) {
		var config = {
		  kitId: 'xeu8jut',
		  scriptTimeout: 3000
		},
		h=d.documentElement,t=setTimeout(function(){h.className=h.className.replace(/\bwf-loading\b/g,"")+" wf-inactive";},config.scriptTimeout),tk=d.createElement("script"),f=false,s=d.getElementsByTagName("script")[0],a;h.className+=" wf-loading";tk.src='//use.typekit.net/'+config.kitId+'.js';tk.async=true;tk.onload=tk.onreadystatechange=function(){a=this.readyState;if(f||a&&a!="complete"&&a!="loaded")return;f=true;clearTimeout(t);try{Typekit.load(config)}catch(e){}};s.parentNode.insertBefore(tk,s)
	  })(document);
	</script>
		
	<!-- Google Analytics -->	
	<script>
	  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

      ga('create', 'UA-86786826-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');

	</script>
	
	<!-- jQuery -->
	<script src="/js/jquery.min.js"></script>
	
	<!-- Bootstrap -->
	<script src="/js/transition.js"></script>
	<script src="/js/collapse.js"></script>


</head>


<body>
	
	<div id="header">
	<nav class="navbar navbar-default navbar-static-top" role="navigation">	 
		<div class="container">	

			<div class="navbar-header">
				<button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar-collapse-element" aria-expanded="false">
					<span class="sr-only">Toggle navigation</span>
					<span class="icon-bar"></span>
					<span class="icon-bar"></span>
					<span class="icon-bar"></span>
				</button>
									
				<a class="navbar-brand" href="/">Taming the BEAST</a>
				<object class="logo navbar-brand" data="/images/logo_bw.svg" type="image/svg+xml"></object>
				
			</div>

			<div class="collapse navbar-collapse" id="navbar-collapse-element">
				<ul class="nav navbar-nav navbar-right">					
					
					<li>
					
					<a href="/news/">news</a></li>

					
                                        <li>
					
                                        <a href="/workshops/">workshops</a></li>

					
					<li>
					
					<a href="/tutorials/">tutorials</a></li>	    	  

					
					<li>
					
					<a href="/contribute/">contribute</a></li>
					<li><a type="application/rss+xml" href="/feed.news.xml"><i class="fa fa-rss"></i></a></li>
				</ul>
			</div>
			
		</div>
	</nav>	
</div>

<!--div class="headerspacer"></div-->

	
	<div class="container">	
		
	<script src="/js/sidebar.js"></script>




<div class="row">
	<div class="hidden-xs hidden-sm col-md-12">
		<div class="bigtitle titlebox">
			<ol class="breadcrumb">
			
			
				
				
				
				<li><a class="off" href="/tutorials/Reassortment-Tutorial/">Estimating Reassortment Networks with CoalRe</a></li>
				
			
			</ol>
		</div>
		<p>
		<div class="head">
			
				Estimating Reassortment Networks with CoalRe
			
		</div>
		
		<div class="smallhead">
			by
			
			
				
						Nicola F. Müller
					
						
					
			
		</div>
		
	</div>				
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-3">
	<div id="sidebar">
		<div class="bigspacer"></div>
		<div class="smallhead">
			Tutorial
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
			<div class="smallspacer"></div>
			<li>
				<i class="fa fa-github fa-fw"></i>
				<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial" target="_blank">Github repository</a>
			</li>
			<div class="smallspacer"></div>
			<li>
				<i class="fa fa-certificate fa-fw"></i>
				<a class="off" href="LICENSE.html">
				License</a>
			</li>
			<div class="smallspacer"></div>
			<li>			
				<i class="fa fa-bar-chart fa-fw"></i>
				<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/graphs/traffic" target="_blank">
				Statistics</a>
			</li>
			</ul>
		</div>		
		<div class="spacer"></div>
		 	
		<div class="smallhead">
			Data
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-align-justify fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/raw/master/data/h1n1pdm_HA.fasta" target="_blank" rel="nofollow">
						h1n1pdm_HA.fasta
					</a>
				</li>
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-align-justify fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/raw/master/data/h1n1pdm_NA.fasta" target="_blank" rel="nofollow">
						h1n1pdm_NA.fasta
					</a>
				</li>
			
			</ul>
		</div>		
		<div class="spacer"></div>
		
		 	
		<div class="smallhead">
			XML
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-code fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/raw/master/xml/H1N1pdm.xml" target="_blank" download rel="nofollow">
						H1N1pdm.xml
					</a>
				</li>
			
			</ul>
		</div>	
		<div class="spacer"></div>
		
		
		 
		<div class="smallhead">
			Output
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-file-text-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/raw/master/precooked_runs/h1n1pdm_HA.log" target="_blank" download rel="nofollow">
						h1n1pdm_HA.log
					</a>
				</li>
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-file-text-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/raw/master/precooked_runs/H1N1pdm.network.trees" target="_blank" download rel="nofollow">
						H1N1pdm.network.trees
					</a>
				</li>
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-file-text-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/raw/master/precooked_runs/h1n1pdm_NA_1,2.trees" target="_blank" download rel="nofollow">
						h1n1pdm_NA_1,2.trees
					</a>
				</li>
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-file-text-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/raw/master/precooked_runs/h1n1pdm_HA_1,2.trees" target="_blank" download rel="nofollow">
						h1n1pdm_HA_1,2.trees
					</a>
				</li>
			
				
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-file-text-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/raw/master/precooked_runs/H1N1pdm.mcc.network.trees" target="_blank" download rel="nofollow">
						H1N1pdm.mcc.network.trees
					</a>
				</li>
			
			</ul>
		</div>		
		
		<div class="spacer"></div>
		<div class="smallhead">
			Contributors
		</div>	
		<div class="pad-left note">
				
			<div class="smallspacer"></div>				
			<div>
				<a class="off" href="https://github.com/nicfel">
    			<img class="pull-left avatar" src="https://avatars0.githubusercontent.com/u/19623728?v=4&s=50">
    			<div class="handlebox" style="padding-left:5px;">
    				nicfel
    			</div>
 				</a>
 			</div>			
 			
		</div>
		<div class="bigspacer"></div>
		<div class="smallhead">
			Latest commits
		</div>		
		<div class="pad-left smallnote">
			<ul class="list-unstyled">
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/commit/29ebe00d8c1651396f5e512b2cbfd0a4f87565a1">
					05 Aug 2019 - <span class="text-gray">Fixed bibliography database link.</span>
					</a> 
				</li>
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/commit/2e42254df393e59e6298fdc24663735e00b4deb8">
					05 Aug 2019 - <span class="text-gray">Fixed typo.</span>
					</a> 
				</li>
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/commit/a61c696560c9acba1df739a28ef818d0f128914c">
					05 Aug 2019 - <span class="text-gray">Added missing ref.</span>
					</a> 
				</li>
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/commit/885e494968d8db89d94ab85e29913893f944d3cf">
					31 Jul 2019 - <span class="text-gray">adds some more details and explanations</span>
					</a> 
				</li>
			
				<div class="smallspacer"></div>
				<li> 
					<i class="fa fa-check-square-o fa-fw"></i>
					<a class="off" href="https://github.com/nicfel/Reassortment-Tutorial/commit/c5234cdda667cecc13e1a30277a9dc12e7759c7e">
					30 Jul 2019 - <span class="text-gray">figure updates etc.</span>
					</a> 
				</li>
			
			</ul>
		</div>		
		<div class="spacer"></div>	
	</div>
	</div>

	<div class="col-md-8">				
		<div class="post">
			<p>
			<h1>Background</h1>

<p>Phylogenetic trees are often used to describe the history of genetic sequences. 
There are however several processes that recombine genetic material.
Different parts of genome can then code for different histories and the shared history of the full genome can not be represented anymore by a tree.
Reassortment is such a process that can reshuffle segments when people are infected by multiple viruses.
In order to allow inference in the presence of reassortment, we introduced the coalescent with reassortment and a MCMC framework to infer ressortment networks, the embedding of segment trees and evolutionary parameters.</p>

<hr>

<h1>Programs used in this Exercise</h1>

<h3>BEAST2 - Bayesian Evolutionary Analysis Sampling Trees 2</h3>

<p>BEAST2 (<a href="http://www.beast2.org">http://www.beast2.org</a>) is a free software package for Bayesian evolutionary analysis of molecular sequences using MCMC and strictly oriented toward inference using rooted, time-measured phylogenetic trees. This tutorial is written for BEAST v2.5.2 <a class="citation" href="#BEAST2book2014">(Drummond &amp; Bouckaert, 2014)</a>.</p>

<h3>BEAUti2 - Bayesian Evolutionary Analysis Utility</h3>

<p>BEAUti2 is a graphical user interface tool for generating BEAST2 XML configuration files.</p>

<p>Both BEAST2 and BEAUti2 are Java programs, which means that the exact same code runs on all platforms. For us it simply means that the interface will be the same on all platforms. The screenshots used in this tutorial are taken on a Mac OS X computer; however, both programs will have the same layout and functionality on both Windows and Linux. BEAUti2 is provided as a part of the BEAST2 package so you do not need to install it separately.</p>

<h3>TreeAnnotator</h3>

<p>TreeAnnotator is used to summarise the posterior sample of trees to produce a maximum clade credibility tree. It can also be used to summarise and visualise the posterior estimates of other tree parameters (e.g. node height).</p>

<p>TreeAnnotator is provided as a part of the BEAST2 package so you do not need to install it separately.</p>

<hr>

<h1>Practical: Setting up an coalescent with reassortment analysis</h1>

<p>The coalescent with reassortment is an approach that allows inferring reassortment networks and rates of segmented viruses from the genetic sequences of individual segments.
In this tutorial, we will describe how to set up a coalecent with reassortment analysis .</p>

<h2>The Data</h2>

<p>After the 2009 influenza A/H1N1 pandemic, this virus become seasonal and is circulating the globe since then.
The dataset we use in this tutorial consists of 25 HA and NA sequences of pandemic 2009 like human influenza A/H1N1 sampled between 2012 and 2017 downloaded from <a href="fludb.org">fludb.org</a>.
The already aligned sequences can be found in the data folder.</p>

<h2>Download CoalRe</h2>

<p>First, we have to download the CoalRe package by using the BEAUti package manager. 
To do so, first open BEAUTi and then go to <code>File &gt;&gt; Manage Packages</code> and download the CoalRe package .</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:50%;" src="figures/Package.png" alt="">
    <figcaption>Figure 1: Download the CoalRe package</figcaption>
</figure>

<p>After the package is installed, re-start BEAUti.</p>

<h2>Load in the sequences</h2>

<p>In order to load in the sequences into BEAUti, they can be dragged and dropped into the paritions window.
Then a window will pop up, where we&#39;ll have to specify that all the sequence files just loaded in are nucleotide sequences.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/LoadSequences.png" alt="">
    <figcaption>Figure 2: Drag and drop the sequence files into the oartitions window.</figcaption>
</figure>

<p>In order to account for rate variations across the different nucleotide sites, we next have to split both alignments into codon positions.
To do so, select one of the segments and the press split and select the 1,2 +3 option. 
This assigns the same relative evolutionary rate to the first 2 codon postitions and a different one for the third position.
Then, repeat the same thing for the other segment.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:30%;" src="figures/SplitAlignments.png" alt="">
    <figcaption>Figure 3: Split alignments into codon positions.</figcaption>
</figure>

<h2>Linking the site models and clock models</h2>

<p>Next, we&#39;ll have to select all partitions, and then press <code>Link Site Models</code>, <code>Link Clock Models</code>.
Linking Clock models leads every partition to have the same evolutionary rate, while linking the site model assigns every partition the same site model.
Assigning every partition the same site model is only done temporarily to speed up the setting up of the xml file and the site models will be unlinked later again.</p>

<h2>Setting up the sampling times</h2>

<p>Next, we&#39;ll have to set up the sampling times. 
To do so, go to <code>Tip Dates</code>  and select <code>Use tip dates</code>
Specify the dates format to be <code>yyyy-M-dd</code> and press the <code>Auto-configure</code> button.
In the window that should pop up, select split on character </p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:30%;" src="figures/SplitCharacter.png" alt="">
    <figcaption>Figure 4: Split character and tke the second group to get the sampling times .</figcaption>
</figure>

<p>In order to set the tip dates for both segments, select both partitions an press <code>OK</code> to clone the tip dates</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/CloneTipDates.png" alt="">
    <figcaption>Figure 5: Clone tip dates .</figcaption>
</figure>

<h2>Setting up the site model</h2>

<p>As a site model, we will use an $HKY+\Gamma_4$ model. 
To do so, first set the site model from <code>JC69</code> to <code>HKY</code>, which allows transition and transversion rates to differ.
Also, set the <code>Gamma Category Count</code> to 4 and make sure to click <code>estimate</code> for the <code>Substitution Rate</code>. 
The last part allows each segment, as well as the first two and the third codon position to have different relative rates of evolution.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/SiteModel.png" alt="">
    <figcaption>Figure 6: Setting up the site models .</figcaption>
</figure>

<p>We next have to go back to the <code>Partitions</code> ta, select all Partitions and then press <code>Unlink Site Models</code>. 
As mentioned above, the linking of the site models was only done to speed up setting up the xml.
After we unlinked the site models, each parition will have a site model that is setup the same way, but that has parameters that can be independently estimated. </p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/UnlinkSiteModels.png" alt="">
    <figcaption>Figure 7: Setting up the site models.</figcaption>
</figure>

<h2>Setting up the Priors</h2>

<p>We can leave the clock model as is, which means that we use a Strict Clock Model and directly go the the <code>Priors</code> tab.
There currently is no other option than a strict clock model anyway when considering networks instead of trees.
The first and most important thing we have to do here, is to to change the <code>Yule Model</code> to <code>Coalescent With Reassortment Constant Population</code>. 
This has to be done for both (!) trees.
This ensures that both segment trees are linked to a network in which they are embedded and that that network is linked to a coalescent with reassortment network prior.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/CoalescentWithReassortment.png" alt="">
    <figcaption>Figure 8: Changing the Yule model to the coalescent with reassortment.</figcaption>
</figure>

<p>We next have to set the prior distribution on the parameters.
The prior distribution on the effective population size can be left as is, but we have to change the prior on the reassortment rate.
Set the prior distribution on the reassortment rate to be an exponential distribution with mean 0.25. 
This means that we assume a priori that on average there is one reassortment event per lineage occurring every 4 years.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/ReassortmentPrior.png" alt="">
    <figcaption>Figure 9: Setting the prior distribution on the reassortment rate.</figcaption>
</figure>

<h2>Setting up the Chain Length</h2>

<p>Next, we can go to the <code>MCMC</code> panel to set the chain length. For this analysis with just a few sequences, a chain length of 5 million should be enough.
This was the last step of setting up the xml and we can now save it by going to <code>File &gt; Save as</code></p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:20%;" src="figures/SaveXml.png" alt="">
    <figcaption>Figure 10: Save xml.</figcaption>
</figure>

<h2>Run the xml</h2>

<p>Next, open <code>BEAST</code> and run the xml. 
This should take somwhere in the order of= 15-20 minutes.
Alternative, the folder <code>precooked_runs</code> contains the log files of the run.</p>

<h2>Inspect the run in Tracer</h2>

<p>Next, we have to check whether everything converged.
To do so, we can opern the program Tracer and load the <code>*.log</code> file.
All ESS values should optimally be above 200.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/Tracer.png" alt="">
    <figcaption>Figure 11: Check convergence in Tracer.</figcaption>
</figure>

<p>Then, we can check what rates were inferred.
The <code>clockRate</code> denotes the average rate of evolution across all segments.
The &#39;reassortmentRateCwR.alltrees` denotes the rate of reassortment (from present to past) per lineage and year.</p>

<h2>Summarize the posterior distribution of networks</h2>

<p>Next, we can summarize the distribution of networks by maximizing the clade credibilities.
To do so, open <code>BEAUti</code> and select <code>File &gt; Launch Apps</code>.
Then, select <code>Reassortment Network Annotator</code>.</p>

<p>Next, choose the <code>networks.trees</code> file as input for the <code>Reassortment Network log file</code> and choose the file where the mcc network should be saved to and press analyse.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/Summary.png" alt="">
    <figcaption>Figure 12: Produce the maximum clade credibility network.</figcaption>
</figure>

<p>The option &#39;Removes segments from the summary&#39; is not relevant for datasets with only 2 segments.
When there are more segments, however, it can e.g. be interesting to look at the reassortment network of pairs of segments.
This option allows to remove segments from an analysis.
Keep in mind though that the numbering of segments is not neccessarily the biological one, but the alphabetical one (for implementation reasons).</p>

<h2>Visualize the network using icytree.org</h2>

<p>Next, open your browser and go to the webpage <a href="icytree.org">icytree.org</a><a class="citation" href="#vaughan2017icytree">(Vaughan, 2017)</a>
The resulting mcc network file can now be drag and dropped into icytree to visualize the network.
Icytree plots the network as a base tree that is connected by dotted branches. 
This implies that at a reassortment event, there is a difference between the two parent branches. 
This is, however, not the case in the coalescent with reassortment model, but for simplicity is plotted like this.
The &quot;main&quot; branch here is the always the parent branch that carries more segments.
If both branches carry the same amount of segments, the branch that is closer the next event is chosen as the main branch.</p>

<figure>
    <a id="fig:example1"></a>
    <img style="width:70%;" src="figures/IcyTree.png" alt="">
    <figcaption>Figure 13: Visualize the mcc network in icytree.</figcaption>
</figure>

<p>The 95 $\%$ highest posterior density intervals for node heights can be plotted by going to <code>Style &gt; Node height error bars</code>.
The posterior support for each node can be shown by going to <code>Style &gt; Internal node text</code>. </p>

<hr>

<h1>Useful Links</h1>

<ul>
<li>BEAST 2 website and documentation: <a href="http://www.beast2.org/">http://www.beast2.org/</a></li>
<li>Join the BEAST user discussion: <a href="http://groups.google.com/group/beast-users">http://groups.google.com/group/beast-users</a></li>
</ul>

<hr>

<h1>Relevant References</h1>

<ol class="bibliography"><li><span id="BEAST2book2014">Drummond, A. J., &amp; Bouckaert, R. R. (2014). <i>Bayesian evolutionary analysis with BEAST 2</i>. Cambridge University Press.</span></li>
<li><span id="vaughan2017icytree">Vaughan, T. G. (2017). IcyTree: Rapid browser-based visualization for phylogenetic trees and networks. <i>Bioinformatics</i>. https://doi.org/10.1093/bioinformatics/btx155</span></li></ol>


			<hr>
			<h1>Citation</h1>

			
<p>If you found <strong>Taming the BEAST</strong> helpful in designing your research, please cite the following paper:</p>

<p style="margin-left: 40px">
Joëlle Barido-Sottani, Veronika Bošková, Louis du Plessis, Denise Kühnert, Carsten Magnus, Venelin Mitov, Nicola F. Müller, Jūlija Pečerska, David A. Rasmussen, Chi Zhang, Alexei J. Drummond, Tracy A. Heath, Oliver G. Pybus, Timothy G. Vaughan, Tanja Stadler (2018). Taming the BEAST – A community teaching material resource for BEAST 2. <em>Systematic Biology</em>, <em>67(1)</em>, 170–-174. doi: <a target="_blank" href="https://doi.org/10.1093/sysbio/syx060">10.1093/sysbio/syx060</a>
</p> 
		</div>
	</div>			
	<div class="col-md-1"></div>
</div>


	
	</div>

	<div class="footerspacer"></div>

<div class="footer">
	<div class="container">
		<div class="col-md-6 reduced-gutter">
			<div class="bigspacer"></div>		
				<ul class="list-inline">
					<li class="footernav">
						<i class="fa"></i> <a class="foot" href="/about/">about</a>
					</li>	
					<li class="footernav">
						<i class="fa"></i> <a class="foot" href="/contact/">contact</a>
					</li>	
					<li class="footernav">
						<i class="fa"></i> <a class="foot" href="/license/">license</a>
					</li>	
				</ul>
			<div class="spacer"></div>			
		</div>
		
		<div class="col-md-5 text-right">
			<div class="bigspacer"></div>		
				<ul class="list-inline">					
					<li class="footernav">
						<i class="fa"></i>powered by <a class="foot" href="http://pages.github.com">GitHub</a> and <a class="foot" href="http://www.jekyllrb.com">Jekyll</a>
					</li>								
				</ul>
			<div class="spacer"></div>			
		</div>

		<div class="col-md-1 reduced-gutter">
			<div class="spacer"></div>						
				<div class="title text-gray">
					<a class="foot" href="http://www.github.com/taming-the-beast"><i class="fa fa-github fa-fw"></i></a>
				</div>
			<div class="spacer"></div>			
		</div>
	</div>
</div>

	
	<!--div id="footer"><span style="display:none">foo</span></div-->
		
</body>
</html>

