<p>% This file was created with JabRef 2.9.2.
% Encoding: MacRoman</p>

<p>Bouckaert, R., Heled, J., Kühnert, D., Vaughan, T., Wu, C.-H., Xie, D., Suchard, M. A., Rambaut, A., &amp; Drummond, A. J. (2014). BEAST 2: a software platform for Bayesian evolutionary analysis. <i>PLoS Computational Biology</i>, <i>10</i>(4), e1003537. https://doi.org/10.1371/journal.pcbi.1003537</p>

<p>Drummond, A. J., &amp; Bouckaert, R. R. (2014). <i>Bayesian evolutionary analysis with BEAST 2</i>. Cambridge University Press.</p>

<p>Gavryushkina, A., Welch, D., Stadler, T., &amp; Drummond, A. J. (2014). Bayesian Inference of Sampled Ancestor Trees for Epidemiology and Fossil Calibration. <i>PLoS Computational Biology</i>, <i>10</i>(12), e1003919. https://doi.org/10.1371/journal.pcbi.1003919</p>

<p>Kühnert, D., Stadler, T., Vaughan, T. G., &amp; Drummond, A. J. (2016). Phylodynamics with migration: a computational framework to quantify population structure from genomic data. <i>Molecular Biology and Evolution</i>, <i>33</i>(8), 2102–2116.</p>

<p>Heath, T. A., Huelsenbeck, J. P., &amp; Stadler, T. (2014). The fossilized birth–death process for coherent calibration of divergence-time estimates. <i>Proceedings of the National Academy of Sciences</i>, <i>111</i>(29), E2957–E2966.</p>

<p>Mulvey, L. P. A., Nikolic, M. C., Allen, B. J., Heath, T. A., &amp; Warnock, R. C. M. (2025). From fossils to phylogenies: exploring the integration of paleontological data into Bayesian phylogenetic inference. <i>Paleobiology</i>, <i>51</i>(1), 214–236. https://doi.org/10.1017/pab.2024.47</p>

<p>Scire, J., Barido-Sottani, J., Kühnert, D., Vaughan, T. G., &amp; Stadler, T. (2022). Robust phylodynamic analysis of genetic sequencing data from structured populations. <i>Viruses</i>, <i>14</i>(8), 1648.</p>

<p>Stadler, T. (2010). Sampling-through-time in birth–death trees. <i>Journal of Theoretical Biology</i>, <i>267</i>(3), 396–404. https://doi.org/10.1016/j.jtbi.2010.09.010</p>
