<p>% This file was created with JabRef 2.9.2.
% Encoding: MacRoman</p>

<p>Ogilvie, H. A., Heled, J., Xie, D., &amp; Drummond, A. J. (2016). Computational Performance and Statistical Accuracy of *BEAST and Comparisons with Other Methods. <i>Systematic Biology</i>, <i>65</i>(3), 381–396. <a href="https://doi.org/10.1093/sysbio/syv118">https://doi.org/10.1093/sysbio/syv118</a></p>

<p>Ogilvie, H. A., Bouckaert, R. R., &amp; Drummond, A. J. (2017). StarBEAST2 Brings Faster Species Tree Inference and Accurate Estimates of Substitution Rates. <i>Molecular Biology and Evolution</i>, <i>34</i>(8), 2101–2114. <a href="https://doi.org/10.1093/molbev/msx126">https://doi.org/10.1093/molbev/msx126</a></p>

<p>Scornavacca, C., &amp; Galtier, N. (2017). Incomplete Lineage Sorting in Mammalian Phylogenomics. <i>Systematic Biology</i>, <i>66</i>(1), 112–120. <a href="https://doi.org/10.1093/sysbio/syw082">https://doi.org/10.1093/sysbio/syw082</a></p>

<p>Slater, G. J. (2015). Iterative adaptive radiations of fossil canids show no evidence for diversity-dependent trait evolution. <i>Proceedings of the National Academy of Sciences</i>, <i>112</i>(16), 4897–4902. <a href="https://doi.org/10.1073/pnas.1403666111">https://doi.org/10.1073/pnas.1403666111</a></p>

<p>Bouckaert, R. R. (2010). DensiTree: making sense of sets of phylogenetic trees. <i>Bioinformatics</i>, <i>26</i>(10), 1372–1373. <a href="https://doi.org/10.1093/bioinformatics/btq110">https://doi.org/10.1093/bioinformatics/btq110</a></p>

<p>Lindblad-Toh, K., Wade, C. M., Mikkelsen, T. S., Karlsson, E. K., Jaffe, D. B., Kamal, M., … others. (2005). Genome sequence, comparative analysis and haplotype structure of the domestic dog. <i>Nature</i>, <i>438</i>(7069), 803–819.</p>

<p>Yang, Z. (1994). Maximum likelihood phylogenetic estimation from DNA sequences with variable rates over sites: Approximate methods. <i>Journal of Molecular Evolution</i>, <i>39</i>(3), 306–314. <a href="https://doi.org/10.1007/BF00160154">https://doi.org/10.1007/BF00160154</a></p>

<p>Hasegawa, M., Kishino, H., &amp; Yano, T. (1985). Dating of the human-ape splitting by a molecular clock of mitochondrial DNA. <i>Journal of Molecular Evolution</i>, <i>22</i>(2), 160–174. <a href="https://doi.org/10.1007/BF02101694">https://doi.org/10.1007/BF02101694</a></p>

<p>Hugall, A. F., Foster, R., Lee, M. S. Y., &amp; Hedin, M. (2007). Calibration Choice, Rate Smoothing, and the Pattern of Tetrapod Diversification According to the Long Nuclear Gene RAG-1. <i>Systematic Biology</i>, <i>56</i>(4), 543–563. <a href="https://doi.org/10.1080/10635150701477825">https://doi.org/10.1080/10635150701477825</a></p>

<p>Prado-Martinez, J., Sudmant, P. H., Kidd, J. M., Li, H., Kelley, J. L., Lorente-Galdos, B., … Marques-Bonet, T. (2013). Great ape genetic diversity and population history. <i>Nature</i>, <i>499</i>, 471–475. Retrieved from <a href="http://dx.doi.org/10.1038/nature12228">http://dx.doi.org/10.1038/nature12228</a></p>

<p>Zrzavý, J., &amp; Řičánková, V. (2004). Phylogeny of Recent Canidae (Mammalia, Carnivora): relative reliability and utility of morphological and molecular datasets. <i>Zoologica Scripta</i>, <i>33</i>(4), 311–333. <a href="https://doi.org/10.1111/j.0300-3256.2004.00152.x">https://doi.org/10.1111/j.0300-3256.2004.00152.x</a></p>

<p>Gavryushkina, A., Welch, D., Stadler, T., &amp; Drummond, A. J. (2014). Bayesian inference of sampled ancestor trees for epidemiology and fossil calibration. <i>PLoS Computational Biology</i>, <i>10</i>(12), e1003919.</p>

<p>Heled, J., &amp; Bouckaert, R. R. (2013). Looking for trees in the forest: summary tree from posterior samples. <i>BMC Evolutionary Biology</i>, <i>13</i>(1), 221. <a href="https://doi.org/10.1186/1471-2148-13-221">https://doi.org/10.1186/1471-2148-13-221</a></p>
